packages/xo-server/src/xapi/mixins/vm.js

```diff
@@ -125,7 +125,10 @@ export default {
       const isHvm = isVmHvm(vm)
 
       if (isHvm) {
-        if (!isEmpty(vdis) || installMethod === 'network') {
+        if (
+          (isEmpty(vdis) && isEmpty(existingVdis)) ||
+          installMethod === 'network'
+        ) {
```

Comment:
There are to changes here, the removal of a negation (`!`) and the addition of `existingVdis`, are both of them intended?

Can you explain them?

Reply:
These changes respond to the [issue](https://github.com/vatesfr/xen-orchestra/issues/4980) requirement.

> only if the VM is HVM and (created without any hard disk OR installed from network), then add network first (or move it first if it was already there: cdn => ncd)

`Created without any hard disks` means that `vdis` and `existingVdis` need to be empty.